### PR TITLE
SREP backplane requirements

### PR DESCRIPTION
**Pull request by jewzaam**
### What type of PR is this?
documentation

### What this PR does / why we need it?
Capture requirements for SREP access and cleans up a few things in the existing backplane RBAC.

### Which Jira/Github issue(s) this PR fixes?
[OSD-17467](https://issues.redhat.com//browse/OSD-17467) 

### Special notes for your reviewer:
Intent is not to change any RBAC for SREP.  Changes in RBAC were done are intended to cleanup duplicates or fix issues only.

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [x] Included documentation changes with PR
- [x] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: deploy/backplane/srep/10-srep-admins-project.ClusterRole.yml [hunk 1]
```diff
@@ -65,14 +65,6 @@ rules:
   - podsecuritypolicysubjectreviews
   verbs:
   - create
-# SRE can review RBAC policies (i.e. oc adm policy who-can ...)
```
Comment (cblecker) [inline deploy/backplane/srep/10-srep-admins-project.ClusterRole.yml:68]: Why are these being removed?

File: docs/backplane/requirements/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Backplane Requirements
```
Comment (cblecker) [inline docs/backplane/requirements/README.md:1]: How does this align with #1587 ?

File: deploy/backplane/srep/10-srep-admins-project.ClusterRole.yml (hunk 1 above)
Comment (jewzaam) [reply deploy/backplane/srep/10-srep-admins-project.ClusterRole.yml:68]: @cblecker commit message: "Removed srep project permissions already granted at cluster scope"

File: docs/backplane/requirements/README.md (hunk 2 above)
Comment (jewzaam) [reply docs/backplane/requirements/README.md:1]: https://github.com/openshift/managed-cluster-config/pull/1587 provides guidelines.  This is govered by SD-ADR-0138, not the guidelines.  Should we capture the ADR in the repo for full context of the workflow?

**Comment by jewzaam** [comment]
/hold

would like to merge https://github.com/openshift/managed-cluster-config/pull/1590 first

File: docs/backplane/requirements/srep/00-cloud-console.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+SREP must be able to access customer cloud account.
+
+Using `ocm backplane cloud console`, has permission to assume the *Support-Role* in the customer's AWS Account or GCP project.  
+
+## AWS
+See [sts_support_permission_policy.json](../../../../resources/sts/4.12/sts_support_permission_policy.json)
```
Comment (cblecker) [inline docs/backplane/requirements/srep/00-cloud-console.md:6]:
```suggestion
See [sts_support_permission_policy.json](/resources/sts/4.12/sts_support_permission_policy.json)
```

File: docs/backplane/requirements/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* Requirements encompass all access requirements including:
+    * conditions for when authorization applies
+    * authorization required in-cluster
+    * authorization required for cloud accounts
+    * authorization required for managed scripts
+    * if privilege escalation is required (note if approved guidelines must be updated)
+
+## Directory Structure
+* root directory is `docs/backplane/requirements/`
+* every persona has a unique sub-directory, i.e. `docs/backplane/requirements/srep` for Layered Product SRE
```
Comment (cblecker) [inline docs/backplane/requirements/README.md:20]: Says LPSRE, but links to SREP?

Comment (jewzaam) [reply docs/backplane/requirements/README.md:20]:
```suggestion
* every persona has a unique sub-directory, i.e. `docs/backplane/requirements/srep` for SRE Platform
```

**Pull request merged**